arch_gen/src/lib.rs

```diff
@@ -0,0 +1,7 @@
+// Copyright 2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+extern crate memory_model;
+
+#[cfg(any(target_arch = "x86", target_arch = "x86_64"))]
+pub mod x86;
```

Comment:
In this case we shouldn't advertise the target_arch to both x86 and x86_64. But anyway, I don't think it makes a big difference here because the Linux headers are the same for both x86 and x86_64 so the autogenerated code will look the same for both architectures.